Add IaC Overview Pages - CORE-908 (#792)
gruntwork-io/docs: _docs-sources/iac/whats-this/index.md

# What is the Infrastructure as Code Library?

> The Gruntwork Infrastructure as Code Library (IaC Library) is a collection of reusable code that enables you to deploy and manage infrastructure quickly and reliably. We've taken the thousands of hours we spent building infrastructure on AWS and condensed all that experience and code into pre-built packages or modules.  

Reviewer: :thinking: one phrase that gave me pause was "reusable code" vs "reusable Terraform modules" - not sure what is more clear or helpful. Technically we have all kinds of code blended together, so your current phrasing is probably more accurate.

The library consists of two types of code: Modules & Services

## Modules

> Modules are reusable code to deploy and manage one piece of infrastructure and each one is a battle-tested, best-practices definition of a piece of infrastructure, such as a VPC, ECS cluster, or an Auto Scaling Group. For more information on modules check out the [Modules page](/iac/whats-this/modules/)

Reviewer: Now is our chance to drop battle-tested and best-practices, if it no longer sparks joy :laughing: 

Seriously, I've seen it so many times across our content, I'm not sure if it's helping anyone or not. Maybe it sounds cool? What do you think?

Author: I agree. I pulled this phrase from existing docs and I'm not the biggest fan of it. I must've blocked it out in my brain, not realizing I used it like 4 times in these 3 files, so I'll pull that out and work on some better phrasing.

## Services

> Services in the service catalog are reusable code that combines multiple modules to configure a service for a specific use case. These are designed for specific use cases and meant to be deployed directly. For more information on the service catalog check out the [Services page](/iac/whats-this/services/)

Reviewer: Missing period in second sentence?

> ## The tools used in the IaC Library

Reviewer 2: ```suggestion
## Tools used in the IaC Library
```

> The Gruntwork IaC Library is designed to be deployed using the following tools:

Reviewer 2: ```suggestion
The Gruntwork IaC Library is deployed using the following tools:
```

# What is the Infrastructure as Code Library?

The Gruntwork Infrastructure as Code Library (IaC Library) is a collection of reusable code that enables you to deploy and manage infrastructure quickly and reliably. It promotes code reusability, modularity, and consistency in infrastructure deployments. We've taken the thousands of hours we spent building infrastructure on AWS and condensed all that experience and code into pre-built packages or modules.  

The library consists of two types of code: Modules & Services

## Modules

Modules are reusable code components that are used to deploy and manage specific pieces of infrastructure. These modules encapsulate the configuration and resource definitions required to create and manage a particular component, such as a VPC, ECS cluster, or an Auto Scaling Group. For more information on modules check out the [Modules page](/iac/whats-this/modules/).

## Services

> Services in the service catalog are reusable code that combines multiple modules from the IaC Library. It simplifies the deployment and management of complex infrastructure configurations and rather than dealing with individual modules and their dependencies, users can directly deploy services tailored for a particular use case. 

Reviewer 2: ```suggestion
Services in the service catalog are reusable code that combines multiple modules from the IaC Library, simplifying the deployment and management of complex infrastructure configurations. Rather than dealing with individual modules and their dependencies, users can directly deploy services tailored for a particular use case. 
```
